components/ui/checkbox.tsx

```diff
@@ -0,0 +1,30 @@
+"use client"
+
+import * as React from "react"
+import * as CheckboxPrimitive from "@radix-ui/react-checkbox"
+import { Check } from "lucide-react"
+
+import { cn } from "@/lib/utils"
+
+const Checkbox = React.forwardRef<
+  React.ElementRef<typeof CheckboxPrimitive.Root>,
+  React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root>
+>(({ className, ...props }, ref) => (
+  <CheckboxPrimitive.Root
+    ref={ref}
+    className={cn(
+      "peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground",
```

Comment:
Yeah, it looks like Prettier doesn't enforce its 80-character default line width for long classnames, like those that Tailwind produces. Found this [stackoverflow](https://stackoverflow.com/questions/75712335/prettier-wont-auto-fix-longer-tailwind-css-class-name-stack-according-to-print) talking about it; it's also discussed here https://github.com/tailwindlabs/tailwindcss/discussions/7763.

[Prettier even tried fixing it then reverted that fix after](https://prettier.io/blog/2021/11/25/2.5.0.html#collapse-html-class-attributes-onto-one-line-11827httpsgithubcomprettierprettierpull11827-by-jlongsterhttpsgithubcomjlongster).

### Solution

One approach I came upon is to use template literals for long class lists. I don't mind it. What's tough is that it'd be something we'd have to self-enforce (since Prettier won't). Essentially, we'll have to use our judgement when it comes to breaking down longer class lists.

For reference, the class list you commented becomes this:
```
className={cn(
  `peer
  h-4
  w-4
  shrink-0
  rounded-sm
  border
  border-primary
  ring-offset-background
  focus-visible:outline-none
  focus-visible:ring-2
  focus-visible:ring-ring
  focus-visible:ring-offset-2
  disabled:cursor-not-allowed
  disabled:opacity-50
  data-[state=checked]:bg-primary
  data-[state=checked]:text-primary-foreground`,
  className,
)}
```

Reply:
sounds good, will note to myself to self-enforce which is fine for now